app/src/main/java/com/vestrel00/daggerbutterknifemvp/AppComponent.java

```diff
@@ -27,4 +30,18 @@
 @Component(modules = AppModule.class)
 interface AppComponent {
     void inject(App app);
+
+    /**
+     * The builder for this component that allows the {@link Application} instance to be provided.
+     * Providing the {@link Application} instance in the {@link AppModule} using and abstract
+     * {@link dagger.Binds} method does not work.
+     */
+    @Component.Builder
+    interface Builder {
+
+        @BindsInstance
+        Builder application(Application application);
```

Comment:
Note that we could use different @Named values to provide the Application Context and Activity Context but providing the `Application` itself is way is simpler and lighter in weight.